elliptic-curve/src/hash2curve/map2curve.rs

```diff
@@ -2,11 +2,9 @@
 /// via a mapping method like Simplified Shallue-van de Woestijne-Ulas
 /// or Elligator
 pub trait MapToCurve {
-    /// The input values representing x and y
-    type FieldElement;
     /// The output point
     type Output;
 
     /// Map a field element into a point
-    fn map_to_curve(u: Self::FieldElement) -> Self::Output;
+    fn map_to_curve(u: Self) -> Self::Output;
 }
```

Comment:
I changed this a bit, instead of being implemented on the projective point it's now implemented on the field element.

Reply:
Makes sense. I'm okay with this change.

Reply:
However, could we implement it as `fn map_to_curve(u: &self) -> Self::Output;`

Reply:
That's similar to how `osswu` works.

Reply:
That makes a lot of sense indeed.